### PR TITLE
Feature/catalog service refactoring

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/Foundation/Commerce/code/ModelInitializers/CartModelBuilder.cs [hunk 1]
```diff
@@ -27,25 +27,27 @@ namespace Wooli.Foundation.Commerce.ModelInitializers
...
         public CartModelBuilder(
-            ICatalogRepository catalogRepository,
+            ICatalogService catalogService,
             IEntityMapper entityMapper,
             ICurrencyProvider currencyProvider)
         {
-            this.catalogRepository = catalogRepository;
+            this.catalogService = catalogService;
             this.entityMapper = entityMapper;
             this.currencyProvider = currencyProvider;
```
Comment (yuya-yuliya) [inline src/Foundation/Commerce/code/ModelInitializers/CartModelBuilder.cs:52]: Add Assert

File: src/Foundation/Connect/code/Builders/Products/BaseProductBuilder.cs [hunk 2]
```diff
@@ -61,7 +61,7 @@ protected TEntity BuildWithoutPrices<TEntity>(Item source) where TEntity : BaseP
             var entity = new TEntity();
             this.Initialize(entity, source);
             this.SetCatalogName(entity);
-
+            entity.StockStatusName = "InStock";
```
Comment (yuya-yuliya) [inline src/Foundation/Connect/code/Builders/Products/BaseProductBuilder.cs:64]: Is everything fail without this line?

File: src/Foundation/Commerce/code/Mappers/Profiles/CatalogProfile.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    [ExcludeFromCodeCoverage]
+    public class CatalogProfile : Profile
+    {
+        public CatalogProfile()
+        {
+            this.CreateMap<Connect.Variant, Variant>()
+                .ForMember(dest => dest.VariantId, opt => opt.MapFrom(src => src.Id));
+
+            this.CreateMap<Connect.Product, Product>();
```
Comment (yuya-yuliya) [inline src/Foundation/Commerce/code/Mappers/Profiles/CatalogProfile.cs:33]: Do we have the same mapping in SearchProfile?

File: src/Foundation/Connect/code/Builders/Products/BaseProductBuilder.cs (hunk 2 above)
Comment (larionovadaria) [reply src/Foundation/Connect/code/Builders/Products/BaseProductBuilder.cs:64]: Yes, add to cart doesn't work without it.

File: src/Feature/Catalog/code/Controllers/ProductController.cs [hunk 4]
```diff
@@ -22,38 +22,39 @@ namespace Wooli.Feature.Catalog.Controllers
...
         public ProductController(
             IVisitorContext visitorContext,
-            ICatalogRepository catalogRepository,
+            ICatalogService catalogService,
             IProductListRepository productListRepository)
         {
             this.visitorContext = visitorContext;
-            this.catalogRepository = catalogRepository;
+            this.catalogService = catalogService;
             this.productListRepository = productListRepository;
```
[the remark this answers is not in the document]
Comment (larionovadaria) [reply src/Feature/Catalog/code/Controllers/ProductController.cs:45]: This controller will be removed after switching to the new search.

File: src/Foundation/Commerce/code/ModelInitializers/CartModelBuilder.cs (hunk 1 above)
Comment (larionovadaria) [reply src/Foundation/Commerce/code/ModelInitializers/CartModelBuilder.cs:52]: This one will be removed after switching to the new checkout.

**Review by yuya-yuliya — APPROVED** [review]

**Review by evgenii-nikulin — APPROVED** [review]

**Pull request merged**